common/static/common/js/discussion/views/discussion_inline_view.js

```diff
@@ -0,0 +1,232 @@
+
+            this.newPostView = new NewPostView({
+                el: this.newPostForm,
+                collection: this.discussion,
+                course_settings: this.course_settings,
+                topicId: discussionId,
+                is_commentable_cohorted: response.is_commentable_cohorted
+            });
+
+            this.newPostView.render();
+
+            this.listenTo(this.newPostView, 'newPost:createPost', this.hideNewPost);
+            this.listenTo(this.newPostView, 'newPost:cancel', this.hideNewPost);
+            this.discussion.on('add', this.addThread);
+
+            this.retrieved = true;
+            this.showed = true;
+
+            if (this.isWaitingOnNewPost) {
+                this.newPostForm.removeClass('is-hidden').focus();
+            }
+
+            // Hide the thread view initially
+            this.$('.inline-thread').addClass('is-hidden');
+        },
+
+        navigateToThread: function(threadId) {
+            var thread = this.discussion.get(threadId);
+            this.threadView = new DiscussionThreadView({
+                el: this.$('.forum-content'),
+                model: thread,
+                mode: 'tab',
+                course_settings: this.course_settings
+            });
+            this.threadView.render();
+            this.listenTo(this.threadView.showView, 'thread:_delete', this.navigateToAllPosts);
+            this.threadListView.$el.addClass('is-hidden');
+            this.$('.inline-thread').removeClass('is-hidden');
+        },
+
+        navigateToAllPosts: function() {
+            // Hide the inline thread section
+            this.$('.inline-thread').addClass('is-hidden');
+
+            // Delete the thread view
+            this.threadView.$el.empty().off();
+            this.threadView.stopListening();
+            this.threadView = null;
+
+            // Show the thread list view
+            this.threadListView.$el.removeClass('is-hidden');
+
+            // Set focus to thread list item that was saved as active
+            this.threadListView.$('.is-active').focus();
+        },
+
+        toggleDiscussion: function() {
+            var self = this;
+
+            if (this.showed) {
+                this.hideDiscussion();
+            } else {
+                this.toggleDiscussionBtn.addClass('shown');
+                this.toggleDiscussionBtn.find('.button-text').text(gettext('Hide Discussion'));
+                if (this.retrieved) {
+                    this.$('section.discussion').removeClass('is-hidden');
+                    this.showed = true;
+                } else {
+                    this.loadDiscussions(this.$el, function() {
+                        self.hideDiscussion();
+                        DiscussionUtil.discussionAlert(
+                            gettext('Error'),
+                            gettext('This discussion could not be loaded. Refresh the page to try again.')
```

Comment:
@alisan617 Thanks for catching the inconsistency. Can we change this to "Refresh the page and try again."